lib/api/core/plugins/pluginsManager.js

```diff
@@ -876,19 +887,30 @@ function triggerPipes(pipes, event, data) {
 }
 
 /**
- * For a specific event, return the corresponding wildcard
+ * For a specific event, return the correspondings wildcards
  * @example
- *  getWildcardEvent('data:create') // return 'data:*'
- * @param {string} event
- * @returns {String} wildcard event
+ *  getWildcardEvents('data:create') // return ['data:*']
+ *  getWildcardEvents('data:beforeCreate') // return ['data:*', 'data:before*']
+ * @param {String} event
+ * @returns {Array<String>} wildcard events
  */
-function getWildcardEvent (event) {
-  const indexDelimiter = event.indexOf(':');
-  if (indexDelimiter !== 1) {
-    return event.substring(0, indexDelimiter+1) + '*';
+function getWildcardEvents (event) {
+  const [scope, name] = event.split(':');
```

Comment:
This leaves events with 3 or more separator out of the picture.

Reply:
```suggestion
  const rawEvents = event.split(':');
  const [name] = rawEvents.pop();
  const [scope] = rawEvents;
```